src/pages/Main/QuizCard.jsx

```diff
@@ -87,208 +93,220 @@ const QuizCard = ({
     setIsFavorite(false);
   };
 
-  const handleResetProgress = () => {
-    console.log('QuizCard progress was reset!');
+  const handleResetProgress = (quiz, auth, quizzes, setQuizzes, setLoading) => {
+    deleteAttemptsForUserAndQuiz(quiz, auth, quizzes, setQuizzes, setLoading);
     handleClose();
   };
 
   return (
     <Grid item key={quiz.id} xs={12} sm={6} md={4}>
-      <Card
-        elevation={1}
-        onClick={(event) => {
-          // Only navigate when clicking on the non-interactive parts of the card
-          if (!event.target.closest('.menu-icon') && cardClickable) {
-            window.location.href = `${QUIZ}/${quiz.id}`;
-          }
-        }}
-        sx={{
-          height: '100%',
-          display: 'flex',
-          padding: '20px 30px',
-          minWidth: '270px',
-          position: 'relative',
-          transition: 'transform 0.2s',
-          cursor: 'pointer',
-          '&:hover': {
-            transform: 'scale(1.05)',
-            /*backgroundColor: customColors.hover,*/
-          },
-        }}
-      >
-        <Grid
-          container
-          direction="column"
-          justifyContent="space-between"
-          alignItems="flex-start"
+      {loading ? (
```

Comment:
Good idea to add loading here!

Surprisingly, it's not always the best user experience to add a loading indicator. There's three different time scales that I'd recommend considering since people generally recognize changes that happen in less than 1/2 second as instant.
1. Change happens within less than 0.5 second: don't show any indication of progress
2. Change happens between 0.5 seconds and 5 seconds: show some sort of loading indicator
3. Change happens in greater than 5 seconds: start a background task and let the user do other things while waiting

Here's a little more detail about what I'm talking about https://uxpickle.com/how-long-will-the-busy-spinner-keep-your-user-waiting/

So in this case I'd recommend not showing a loading indicator since the reset progress happens so quickly. A snackbar message will indicate to the user that their action was successful.

Reply:
Super useful to think about.  User error and loading delays are so ubiquitous and I really hadn't thought much about them until this practicum. Such useful concepts to be aware of, and know best practices for, thanks Eli!